modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/ModuleExtPlugin.java

```diff
@@ -0,0 +1,292 @@
+
+		Copy copy = GradleUtil.addTask(
+			project, UNZIP_MODULE_EXT_RESOURCES_TASK_NAME, Copy.class);
+
+		copy.from(
+			new Callable<FileTree>() {
+
+				@Override
+				public FileTree call() {
+					return project.zipTree(configuration.getSingleFile());
+				}
+
+			});
+
+		copy.exclude("**/*.class", "**/MANIFEST.MF");
+		copy.setIncludeEmptyDirs(false);
+
+		SourceSet sourceSet = GradleUtil.getSourceSet(
+			project, SourceSet.MAIN_SOURCE_SET_NAME);
+
+		SourceSetOutput sourceSetOutput = sourceSet.getOutput();
+
+		copy.setDestinationDir(sourceSetOutput.getResourcesDir());
+
+		Task processResources = GradleUtil.getTask(
+			project, JavaPlugin.PROCESS_RESOURCES_TASK_NAME);
+
+		processResources.dependsOn(copy);
+
+		return copy;
+	}
+
+	private void _configureDeployTask(Project project, Jar jar) {
+		final LiferayExtension liferayExtension = GradleUtil.getExtension(
+			project, LiferayExtension.class);
+
+		Copy copy = (Copy)GradleUtil.getTask(
+			project, LiferayBasePlugin.DEPLOY_TASK_NAME);
+
+		copy.from(jar);
+
+		liferayExtension.setDeployDir(
+			new Callable<File>() {
+
+				@Override
+				public File call() throws Exception {
+					return new File(
+						liferayExtension.getAppServerParentDir(),
+						"osgi/marketplace/override");
+				}
+
+			});
+	}
+
+	private Jar _configureJarTask(Project project, final Copy copy) {
+		final Jar jar = GradleUtil.addTask(
+			project, JavaPlugin.JAR_TASK_NAME, Jar.class, true);
+
+		jar.dependsOn(JavaPlugin.CLASSES_TASK_NAME, copy);
+
+		SourceSet sourceSet = GradleUtil.getSourceSet(
+			project, SourceSet.MAIN_SOURCE_SET_NAME);
+
+		SourceSetOutput sourceSetOutput = sourceSet.getOutput();
+
+		jar.from(sourceSetOutput.getClassesDir());
+
+		jar.from(sourceSetOutput.getResourcesDir());
+
+		jar.manifest(
+			new Closure<Void>(project) {
+
+				@SuppressWarnings("unused")
+				public void doCall(Manifest manifest) {
+					File file = new File(
+						copy.getTemporaryDir(), "META-INF/MANIFEST.MF");
+
+					if (!file.exists()) {
+						System.out.println("MANIFEST.MF not found");
+
+						return;
+					}
+
+					try (FileInputStream fileInputStream =
+							new FileInputStream(file)) {
+
+						Map map = new HashMap<>();
+
+						java.util.jar.Manifest originalManifest =
+							new java.util.jar.Manifest(fileInputStream);
+
+						Attributes attributes =
+							originalManifest.getMainAttributes();
+
+						for (Object key : attributes.keySet()) {
+							Name name = (Name)key;
+
+							String value = attributes.getValue(name);
+
+							String nameString = name.toString();
+
+							if (nameString.equals("Bundle-Version")) {
+								Version version = Version.parseVersion(value);
+
+								String qualifier = version.getQualifier();
+
+								String suffix = "ext";
+
+								if (qualifier.isEmpty()) {
+									value += '.' + suffix;
+								}
+								else {
+									value += '-' + suffix;
+								}
+							}
+
+							map.put(nameString, value);
+						}
+
+						jar.setArchiveName(
```

Comment:
I don't like setting something else in a closure where you're calculating the value of `manifest`. Please move it outside.